src/process/mod.rs

```diff
@@ -679,6 +681,24 @@ pub struct FDInfo {
 }
 
 impl FDInfo {
+    /// Gets a file descriptor from a raw fd
+    pub fn from_raw_fd(pid: pid_t, raw_fd: i32) -> ProcResult<Self> {
+        Self::from_raw_fd_with_root("/proc", pid, raw_fd)
+    }
+
+    /// Gets a file descriptor from a raw fd based on a specified `/proc` path
+    pub fn from_raw_fd_with_root(root: impl AsRef<Path>, pid: pid_t, raw_fd: i32) -> ProcResult<Self> {
+        let path = root.as_ref().join(pid.to_string()).join("fd").join(raw_fd.to_string());
+        let link = wrap_io_error!(path, read_link(&path))?;
+        let md = wrap_io_error!(path, path.metadata())?;
```

Comment:
This all looks good, except this part should be `path.symlink_metadata()`, so that the behavior here is consistent with what we're doing in `Process::fd()` (which uses [DirEntry::metadata()](https://doc.rust-lang.org/nightly/std/fs/struct.DirEntry.html#method.metadata) which doesn't dereference the symlink)

Reply:
Fixed by 78f3385